avocado_vt/vt_agent/src/avocado_vt/agent/core/logger.py

```
@@ -0,0 +1,183 @@
+    :param format_str: Log format string
+    :type format_str: str
+    :return: Configured file handler
+    :rtype: logging.handlers.RotatingFileHandler
+    :raises OSError: If log file cannot be created
+    """
+    log_dir = os.path.dirname(log_file)
+    if not os.path.exists(log_dir):
+        try:
+            os.makedirs(log_dir, exist_ok=True)
+        except OSError as e:
+            raise OSError(f"Cannot create log directory {log_dir}: {e}") from e
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)